### PR TITLE
Kernel: Send a SIGCHLD to the parent upon suspension of a process that is controlling a TTY

### PR DESCRIPTION
**Pull request by alimpfard**
Fixes the ^Z issue...again.

@nico your idea - send `SIGCHLD` to parent upon suspension.

### CODE REVIEW

**Comment by awesomekling** [comment]
Hey that makes sense

File: Kernel/TTY/TTY.cpp [hunk 1]
```diff
@@ -155,6 +155,9 @@ void TTY::emit(u8 ch)
         if (ch == m_termios.c_cc[VSUSP]) {
             dbg() << tty_name() << ": VSUSP pressed!";
             generate_signal(SIGTSTP);
+            auto process = Process::from_pid(m_pgid);
```
Comment (awesomekling) [inline Kernel/TTY/TTY.cpp:158]: What if this returns `nullptr`

Comment (alimpfard) [reply Kernel/TTY/TTY.cpp:158]: _can_ that return `nullptr`?

AFAIK the pgid has to be a valid pid?

Comment (awesomekling) [reply Kernel/TTY/TTY.cpp:158]: Valid when setting yeah. But just looking at this I get suspicious that maybe we're not clearing `m_pgid` when a process dies.

Comment (alimpfard) [reply Kernel/TTY/TTY.cpp]: Makes sense.

**Pull request merged**